python/ray/tune/suggest/ax.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+try:
+    import ax.service.ax_client as ax_client
+except ImportError:
+    ax_client = None
+
+from ray.tune.suggest.suggestion import SuggestionAlgorithm
+
+
+class AxSearch(SuggestionAlgorithm):
+    """A wrapper around Ax to provide trial suggestions.
+
+    Requires Ax to be installed.
+    Ax is an open source tool from Facebook for configuring and
+    optimizing experiments. More information can be found in https://ax.dev/.
+
+    Parameters:
+        parameters (list[dict]): Parameters in the experiment search space.
+            Required elements in the dictionaries are: "name" (name of
+            this parameter, string), "type" (type of the parameter: "range",
+            "fixed", or "choice", string), "bounds" for range parameters
+            (list of two values, lower bound first), "values" for choice
+            parameters (list of values), and "value" for fixed parameters
+            (single value).
+        objective_name (str): Name of the metric used as objective in this
+            experiment. This metric must be present in `raw_data` argument
+            to `log_data`.
```

Comment:
This metric must also be present in the dict reported/returned by the Trainable.